lib/msal-common/test/crypto/PopTokenGenerator.spec.ts

```diff
@@ -120,7 +116,13 @@ describe("PopTokenGenerator Unit Tests", () => {
                 scopes: TEST_CONFIG.DEFAULT_GRAPH_SCOPE,
                 correlationId: TEST_CONFIG.CORRELATION_ID,
             };
-            sinon.stub(TimeUtils, "nowSeconds").returns(currTime);
+            timeUtilsSpy = jest
+                .spyOn(TimeUtils, "nowSeconds")
+                .mockReturnValue(currTime);
+        });
+
+        afterAll(() => {
```

Comment:
Suggest setting mocks on a per-test basis rather than a per-block basis. It can be really easy to get yourself into trouble with mocks leaking across tests if they're not cleaned up after each test leading to unexpected results.

For simplicity I would suggest setting an afterEach in the topmost describe of each test file which restores all mocks.

Reply:
I agree with you about the simplicity, however I've been thinking in terms of efficiency. Creating a base mock once instead of for each test. Does this affect performance? Probably not? I can change it if you want and everyone prefers simplicity.

Reply:
My #1 priority is ensuring mocks don't unexpectedly remain set from one test to another (this would happen if you set one-off mocks inside a specific test but restore in an afterAll). Simplicity is a nice-to-have but secondary. Performance is a non-concern, the test suite runs in just a few seconds as it is. 

Change beforeAll -> beforeEach and afterAll -> afterEach and it should behave the same as what you've got but safer.